rust/datafusion/src/execution/context.rs

```diff
@@ -512,6 +573,12 @@ pub struct ExecutionConfig {
     optimizers: Vec<Arc<dyn OptimizerRule + Send + Sync>>,
     /// Responsible for planning `LogicalPlan`s, and `ExecutionPlan`
     query_planner: Arc<dyn QueryPlanner + Send + Sync>,
+    /// Default catalog name for table resolution
+    default_catalog: String,
```

Comment:
Would we ever need to distinguish between `""` and `None`? If so, perhaps we can make this field  `default_catalog: Option<String>` instead

Reply:
This is a very good question!

I think we could allow for that distinction, but it gets a little bit messy as we'd probably want to enforce that you can't have a default schema without a default catalog, so it'd look something like `Option<(String, Option<String>)>` (in fact, I had something like that for a bit before I settled on just using `datafusion.public` as the default setting for back-compat).

I'm struggling to think of a use case though 🤔 Maybe if someone were developing a system where they wanted to _force_ users to fully qualify their table names?